pkg/api/resource_types.go

```diff
@@ -60,6 +63,15 @@ func (d *Resource) GetDeletionTimestamp() *metav1.Time {
 	return &metav1.Time{Time: d.Meta.DeletedAt.Time}
 }
 
+func (d *Resource) HasManifestBundle() bool {
+	if d.Manifest != nil {
+		if kind, ok := d.Manifest["kind"]; ok && kind == "ManifestWork" {
```

Comment:
inspecting the payload for exactly and only one kind of resource smells to me because it's implicit.  if Maestro wishes to support _many_ resources in 1 API call, perhaps that should be part of the official API/openapi?

Reply:
Apologies for any confusion. Currently, Maestro still only supports a single resource per API call.
This functionality in this PR is intended solely for internal use, specifically when Maestro accepts gRPC calls. Each API call should include a `manifestwork`, which, in turn, contains a bundle.
However, it's important to note that each API call is designed to handle only one manifestwork.

Reply:
Yes, I understand the desire to go from 1 to Many objects in the payload. I'm suggesting it's better to have that explicit in the API than an implicit string search like this.

Specifically, the Resource struct has the single `Manifest` attribute on it.  In order to make this Many, perhaps a ResourceSet object is the correct one to introduce and it specifically has either many Resources (our Kind) or many Manifests.